### DISCUSSION TITLE
LXQt apps don't clearly specify their preference to Server Side Decorations on Wayland

### DISCUSSION THREAD

**Discussion by TheESN** [lxqt/lxqt#2537] [Ideas]
![311673246-5c39c81a-0d26-4a07-b195-ae659785372d](https://github.com/lxqt/lxqt/assets/102585253/35dcd2a1-816a-4437-8a7f-335272be26bc)
After experiments with pre-alpha of upcoming COSMIC desktop enviroment, I noticed that LXQt apps draw Client Side Decorations despite the fact that COSMIC supports Server Side Decorations. After disscussion with developers we found out that LXQt applications use default mode for xdg-decoration and in COSMIC's case it's CSD. So to resolve this issue LXQt should specify in xdg-decoration that they prefer SSD, so COSMIC could draw them

**Comment by tsujan** [comment]
It's not the job of LXQt. The Wayland compositor decides that.

**Reply by stefonarch** [reply]
FIY there is an issue with CDS, didn't test if that is solved with qt6:
https://github.com/WayfireWM/wayfire/issues/1723

**Reply by TheESN** [reply]
COSMIC compositor expects appliactions to select between CSD and SSD. What should it choose for LXQt apps in your opinion in this case then?

**Reply by tsujan** [reply]
LXQt apps are just Qt apps. COSMIC needs to find a way to deal with Qt.

**Reply by TheESN** [reply]
So the report should be sent to Qt itself?

**Reply by tsujan** [reply]
I don't know. What I know is that a Wayland compositor like LabWC or Wayfire has an option to use SSD or CSD with non-GTK apps. Maybe COSMIC needs to have such an option.

**Reply by tsujan** [reply]
My Qt windows under LabWC:

![labwc](https://github.com/lxqt/lxqt/assets/981076/c504499a-e0c2-4119-b579-8d53786dc8dd)

**Reply by TheESN** [reply]
COSMIC has this option but the issue is that LXQt apps call "unset" through xdg-decoration so it doens't specify preference to compositor so it does what it does by default. To draw SSD app should set it through xdg-decoration (which can be ignored by compositor like GNOME that doesn't have SSD)

**Reply by tsujan** [reply]
> LXQt apps call...

LXQt doesn't call it.

Anyway, other Wayland compositors have solved it; so, COSMIC might too; of course, if it does not depend on GNOME for Wayland.

**Reply by TheESN** [reply]
COSMIC does not depend on GNOME anymore

**Reply by ryanabx** [reply]
Just to clarify: LXQT does use set_mode() to describe their preference to SSDs? https://wayland.app/protocols/xdg-decoration-unstable-v1#zxdg_toplevel_decoration_v1:request:set_mode

**Reply by stefonarch** [reply]
>Wayland compositor like LabWC or Wayfire have an option to use SSD or CSD with non-GTK apps

Actually SSD works also for GTK apps.

The issue with CSD in Qt is still the same I see, so it's completely unusable under wayland: every time you open the Qt-filedialog it will shrink.

**Reply by tsujan** [reply]
> Just to clarify: LXQT does use...

LXQt doesn't.

**Reply by ryanabx** [reply]
> > Just to clarify: LXQT does use...
> 
> 
> 
> LXQt doesn't.

Ok! Thanks for the info

**Reply by tsujan** [reply]
To prevent any misunderstanding as far as possible, I say it again:

LXQt just uses Qt. LXQt doesn't set or call any Wayland function by itself; Qt may do it. Of course, LXQt's git branches and PRs make use of Wayland methods, but (1) none of them is about SSD/CSD, and (2) all of them are provided by qt6-wayland and layer-shell-qt.

If I were you, I'd discuss it with Qt devs (I'm not sure if there's already a discussion somewhere).

**Reply by ryanabx** [reply]
> If I were you, I'd discuss it with Qt devs (I'm not sure if there's already a discussion somewhere).

I'm not personally interested in the issue of CSD vs SSD, but OP is someone who had an issue with lack of SSD in COSMIC with LXQT apps and I wanted to make sure it wasn't a problem on our end! I appreciate the response

**Reply by tsujan** [reply]
> someone who had an issue with lack of SSD in COSMIC with LXQT apps

Then they should have the same "issue" with all Qt apps.

**Reply by Sunderland93** [reply]
I think that in order for LXQt applications to set their own window decoration mode (SSD or CSD), they must directly implement support for the xdg-decoration protocol, which in my opinion does not make any sense, since in this case this is a Qt task.
